### PR TITLE
remove k8s version pin to 1.25 for MEC e2e test

### PR DESCRIPTION
**Pull request by nojnhuh**
**What type of PR is this?**

/kind failing-test

**What this PR does / why we need it**:

The `edgezone` e2e test is failing, and I thought I saw somewhere that it's because the v1.25.0 image that this was pinned to before no longer exists but now I can't seem to find evidence of that. Regardless, v1.25 is EOL so we shouldn't be pinning to it anymore. I couldn't find any commentary on #2836 as to the reason for the 1.25 pin originally.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

- [X] cherry-pick candidate <-- the test is also failing on the release branches

**TODOs**:

- [X] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by mboersma — APPROVED** [review]
/lgtm

**Comment by nojnhuh** [comment]
> The platform image 'cncf-upstream:capi:ubuntu-2204-gen1:127.14.20240517' is not available. Verify that all fields in the storage profile are correct. For more details about storage profile information, please refer to https://aka.ms/storageprofile

@mboersma It looks like this is complaining that the image doesn't exist specifically within this EdgeZone. I wonder if that's why the k8s version was originally pinned. Is there an easy way to copy the existing images to that particular region?

I see that this command works from the CI sub:
```
az vm image show --location canadacentral --urn cncf-upstream:capi:ubuntu-2204-gen1:127.14.20240517
```

But this one which includes the EdgeZone doesn't:
```
az vm image show --location canadacentral --urn cncf-upstream:capi:ubuntu-2204-gen1:microsoftvancouver1:127.14.20240517
```

**Comment by mboersma** [comment]
Based on [this comment](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2836#discussion_r1094982957) I'm inferring that some out-of-band process had to happen to make that particular image available in the testing EdgeZone.

@upxinxin is that correct? Do you know how we might make a newer CAPZ image available to this e2e test introduced in #2836?

(BTW, I checked, and the 1.25.x images were all deprecated programmatically on May 25.)

File: templates/test/ci/cluster-template-prow-edgezone.yaml [hunk 1]
```diff
@@ -154,7 +154,8 @@ metadata:
 spec:
   clusterName: ${CLUSTER_NAME}
   replicas: ${WORKER_MACHINE_COUNT:=2}
-  selector: {}
+  selector:
+    matchLabels: null
```
Comment (willie-yao) [inline templates/test/ci/cluster-template-prow-edgezone.yaml:158]: Why is this change needed?

Comment (nojnhuh) [reply templates/test/ci/cluster-template-prow-edgezone.yaml:158]: That's what happened to get generated ¯\\\_(ツ)\_/¯

Comment (willie-yao) [reply templates/test/ci/cluster-template-prow-edgezone.yaml:158]: That's interesting.. I wonder what change prompted that 🤔

**Comment by mboersma** [comment]
@upxinxin @mogliang the Kubernetes 1.25 image this test required has been deprecated. Do you know how we can get make a more recent CAPZ image available in the right Edge Zone so this test passes again?

**Comment by mboersma** [comment]
My `Skip()` somehow still failed? Oops, let me try again.

**Comment by garogarabedyan** [comment]
Please disable this e2e test.

**Comment by nojnhuh** [comment]
> Please disable this e2e test.

#4900

**Comment by jackfrancis** [comment]
looks like we can close this PR?

**Comment by nojnhuh** [comment]
@garogarabedyan Can we keep this test as long as we migrate off of the `microsoftvancouver1` edge zone onto a different one?

**Comment by nojnhuh** [comment]
@jackfrancis In order to run this test going forward, we need to at least change the edge zone that the test is running in and also update the VM image used once we make it available in that new region. I think we can do those things in this PR yet.

**Comment by garogarabedyan** [comment]
> @garogarabedyan Can we keep this test as long as we migrate off of the `microsoftvancouver1` edge zone onto a different one?

I think you can keep the source code of the test. I think you should not spend time to fix the test before disabling it.

**Comment by nojnhuh** [comment]
Closing until we can reenable the test.

/close
